### PR TITLE
Recommendations - insights

### PR DESCRIPTION
**Pull request by RobertBeaudenon**
Issue #66

- Fetched Recommendations
- Fetched Insights
- Created tabs in audit frontend to navigate between list of audits, insights and recommendations

Screen Shot 2021-04-08 at 11 33 53 AM
Screen Shot 2021-04-08 at 11 34 08 AM
Screen Shot 2021-04-08 at 11 34 31 AM

### CODE REVIEW

File: client/src/app/services/dashboard/dashboard.service.ts [hunk 1]
```diff
@@ -19,4 +19,8 @@ export class DashboardService {
   getRecommendations(params: HttpParams): Observable<any> {
     return this.http.get(`${BASEURL}/recommendation/`, { params });
   }
+
+  getInsights(params: HttpParams): Observable<any> {
+    return this.http.get(`${BASEURL}/insights/`, { params });
+  }
```
Comment (Amanda-Kolopa) [inline client/src/app/services/dashboard/dashboard.service.ts:25]: These services should be moved to audits/manage-audits.service.ts

File: server/audit/views.py [hunk 2]
```diff
@@ -423,20 +426,124 @@ def get_permissions(self):
...
+
+
+def get_values(list_of_dict, key):
+    return [val[key] for val in list_of_dict]
+
+
+def get_random_items(all_bins, all_parts, org_id, frequency_of_bins_or_parts):
+    random_bins_or_parts = random.sample(["bins", "parts"], 1)[0]
+    if random_bins_or_parts == "bins":
+        print('bins')
```
Comment (Amanda-Kolopa) [inline server/audit/views.py:469]: Is this required? 
If not, please remove it

File: server/audit/views.py [hunk 3]
```diff
@@ -423,20 +426,124 @@ def get_permissions(self):
...
+        random_bin = random.sample(all_bins, frequency_of_bins_or_parts)  # increase if we want more items
+        return list(
+            Item.objects.filter(organization=org_id, Bin__in=random_bin).values(
+                'Item_Id'))
+    else:
+        print('part')
```
Comment (Amanda-Kolopa) [inline server/audit/views.py:475]: Is this required? 
If not, please remove it

**Review by Amanda-Kolopa — CHANGES_REQUESTED** [review]
Overall, great work! :smile: 

Notes:

- Each recommendation table should provide the affiliated information of the listed items or bins
-  The arrows on the expanded panel do not correlate with the rest of the UI. For example:
image

File: server/audit/views.py (hunk 2 above)
Comment (RobertBeaudenon) [reply server/audit/views.py:469]: fixed

File: server/audit/views.py (hunk 3 above)
Comment (RobertBeaudenon) [reply server/audit/views.py:475]: fixed

File: client/src/app/services/dashboard/dashboard.service.ts (hunk 1 above)
Comment (RobertBeaudenon) [reply client/src/app/services/dashboard/dashboard.service.ts:25]: fixed

**Review by Amanda-Kolopa — CHANGES_REQUESTED** [review]
Thank you for the changes!

Based on the feedback from our stakeholder meeting, the following changes need to be applied:

- Remove the recommendation _Random Items to audit again_
- Remove table columns _Location_, _Bin_, _Zone_, and _Aisle_ from recommendations _By Frequently Lost Items_ and _By Criticality_
- Add a button under the recommendation table of  _By Frequently Audited Parts_ which will initiate an audit creation process
    - The _Apply_ button should redirect the user to the _Create Template_ page where the recommended items are processed onto the _Selected Fields_ component
    - Once the template is created, the user can select it from the _List of Templates_ and begin the audit creation process

**Review by Amanda-Kolopa — CHANGES_REQUESTED** [review]
Awesome work on the recommendations table! :smile:

Last small change:

- When we initiate one of the recommendations, the created template contains duplicate data
image

**Comment by NicholasNagy** [comment]
Alright @Amanda-Kolopa , I should have fixed it, and it should be good to go!

Thank you! :)

**Review by Amanda-Kolopa — APPROVED** [review]
All looks good now!
Time to merge :smile:

**Pull request merged**